packages/circuit-ui/components/Anchor/Anchor.tsx

```diff
@@ -61,34 +70,67 @@ export type AnchorProps = BaseProps & LinkElProps & ButtonElProps;
  */
 export const Anchor = forwardRef(
   (
-    { className, ...props }: AnchorProps,
+    {
+      className,
+      isExternal,
+      externalLabel,
+      'aria-describedby': descriptionId,
+      children,
+      ...props
+    }: AnchorProps,
     ref?: BaseProps['ref'],
   ): ReturnType => {
     const components = useComponents();
     const Link = components.Link as AsPropType;
+    const externalLabelId = useId();
+    const descriptionIds = clsx(
+      externalLabel && externalLabelId,
+      descriptionId,
+    );
+    const isExternalLink = isExternal || props.target === '_blank';
 
     if (!props.href && !props.onClick) {
-      return <Body as="span" {...props} ref={ref} />;
+      return (
+        <Body as="span" {...props} ref={ref}>
+          {children}
+        </Body>
+      );
     }
 
     if (props.href) {
       return (
         <Body
           {...props}
+          aria-describedby={descriptionIds}
           className={clsx(classes.base, utilClasses.focusVisible, className)}
           as={Link}
           ref={ref}
-        />
+        >
+          {children}
+          {isExternalLink && externalLabel && (
+            <span id={externalLabelId} className={utilClasses.hideVisually}>
+              {externalLabel}
+            </span>
+          )}
```

Comment:
By nesting the external label inside the element, it is announced twice: once as part of the accessible name and once as the accessible description.